File: ABC147/D_Xor_Sum_4/submission1.scala
// AC

object Main extends App {

  val m = 1000000007;

  val sc = new java.util.Scanner(System.in);
  val n = sc.nextInt();
  val as = Array.fill(n)(sc.nextLong());

  val answer = (0 until 60).map { bit =>
    val count = (0 until n).map { i =>
      ((as(i) >> bit) & 1).toInt;
    }.sum;
    val u = (1L << bit) % m;
    u * count % m * (n - count) % m;
  }.sum % m;

  println(answer);
}

File: ABC179/E_Sequence_Sum/submission1.java
// WA

import java.util.Scanner;

class Main {
  public static void main(String[] args) {
    var sc = new Scanner(System.in);
    long n = sc.nextLong();
    int x = sc.nextInt();
    int m = sc.nextInt();

    long answer = 0;
    long x2 = x;
    long m1 = m - 1;
    if (n < 2 * m1) {
      for (int i = 0; i < n; i++) {
        answer += x2;
        x2 = x2 * x2 % m;
      }
    } else {
      for (int i = 0; i < m1; i++) {
        answer += x2;
        x2 = x2 * x2 % m;
      }
      long answer2 = 0;
      for (int i = 0; i < m1; i++) {
        answer2 += x2;
        x2 = x2 * x2 % m;
      }
      answer += answer2 * ((n / m1) - 1);
      int n2 = (int)(n % m1);
      for (int i = 0; i < n2; i++) {
        answer += x2;
        x2 = x2 * x2 % m;
      }
    }

    System.out.println(answer);
  }
}
